src/components/new-safe/create/__tests__/useSyncSafeCreationStep.test.ts

```diff
@@ -39,6 +39,20 @@ describe('useSyncSafeCreationStep', () => {
     expect(mockPushRoute).toHaveBeenCalledWith({ pathname: AppRoutes.welcome.index, query: undefined })
   })
 
+  it('should not go to the first step if no wallet is connected but there is a pending safe', async () => {
```

Comment:
This test case is a duplicate of "should go to the fourth step if there is a pending safe"
We could add the expectation not to call mockPushRoute to that test case.